drivers/flash/flash_hp_ra.c

```diff
@@ -75,10 +47,29 @@ void bgo_callback(flash_callback_args_t *p_args)
 	}
 }
 
-static bool flash_ra_valid_range(off_t area_size, off_t offset, size_t len)
+static bool flash_ra_valid_range(struct flash_hp_ra_data *flash_data, off_t offset, size_t len)
 {
-	if ((offset < 0) || (offset >= area_size) || ((area_size - offset) < len)) {
-		return false;
+	if (flash_data->FlashRegion == DATA_FLASH) {
+		if ((offset < 0) || (offset >= flash_data->area_size) ||
+		    (flash_data->area_size < len + offset)) {
+			return false;
+		}
+	} else {
+#if defined(CONFIG_DUAL_BANK_MODE)
+		if ((offset < 0) || (offset >= FLASH_HP_CF_DUAL_HIGH_END_ADDRESS) ||
+		    (offset >= FLASH_HP_CF_DUAL_LOW_END_ADDRESS &&
+		     offset < FLASH_HP_BANK2_OFFSET) ||
+		    ((len + offset) > FLASH_HP_CF_DUAL_HIGH_END_ADDRESS) ||
+		    ((len + offset) > FLASH_HP_CF_DUAL_LOW_END_ADDRESS &&
+		     (len + offset) < FLASH_HP_BANK2_OFFSET)) {
+			return false;
+		}
+#else
+		if ((offset < 0) || (offset >= flash_data->area_size) ||
+		    (flash_data->area_size < len + offset)) {
```

Comment:
Since you have already checked offset to be in range of area_size it would be better to do the comparison as 'flash_data->area_size - offset < len'.

Reply:
Thank you. I’ve updated it as well.